packages/react-async/README.md

```diff
@@ -0,0 +1,147 @@
+# `@shopify/react-async`
+
+[![Build Status](https://travis-ci.org/Shopify/quilt.svg?branch=master)](https://travis-ci.org/Shopify/quilt)
+[![License: MIT](https://img.shields.io/badge/License-MIT-green.svg)](LICENSE.md) [![npm version](https://badge.fury.io/js/%40shopify%2Freact-async.svg)](https://badge.fury.io/js/%40shopify%2Freact-async.svg) [![npm bundle size (minified + gzip)](https://img.shields.io/bundlephobia/minzip/@shopify/react-async.svg)](https://img.shields.io/bundlephobia/minzip/@shopify/react-async.svg)
+
+Tools for creating powerful, asynchronously-loaded React components.
+
+## Installation
+
+```bash
+$ yarn add @shopify/react-async
+```
+
+## Usage
+
+### `createAsyncComponent()`
+
+`createAsyncComponent` is a function for creating components that are loaded asynchronously on initial mount. However, the resulting component does more than just help you split up your application along component lines; it also supports customized rendering for loading, and creates additional components for smartly preloading or preloading the component’s bundle. Best of all, in conjunction with the Babel and Webpack plugins provided by [`@shopify/async`](https://github.com/Shopify/quilt/tree/master/packages/async), you can easily extract the bundles needed to render your application during server side rendering.
+
+To start, import the `createAsyncComponent` function. The simplest use of this function requires just a `load` function, which returns a promise for a component:
+
+```tsx
+import {createAsyncComponent} from '@shopify/react-async';
+
+const MyComponent = createAsyncComponent({
+  load: () => import('./MyComponent'),
```

Comment:
`renderLoading` is meant for the skeleton fallback components, right? Might be worth including that in this examples